src/curl.c

```diff
@@ -52,6 +52,12 @@ static int curr_version = -1;
 static int req_version = -1;
 static bool resume_download_enabled = true;
 
+/* these are used to handle alternative trust locations */
+#define CA_PATH_NO_MAX 16
+static char *capath;
+static char *fallback_capaths[CA_PATH_NO_MAX];
+static char fallback_capaths_no = 0;
```

Comment:
I think just changing this to 1 will allow you to remove some special case code, see other comments.